static/js/libs/events.js

```diff
@@ -0,0 +1,64 @@
+const Events = {
+  constructor: function() {
+    this.events = {};
+    this.availableHandles = [];
+
+    return this;
+  },
+
+  _addListener: function(type, bindWindow) {
+    const bindTarget = bindWindow ? window : document.body;
+    bindTarget.addEventListener(type, this._handleEvent.bind(this, type));
+  },
+
+  _handleEvent: function(type, event) {
+    const eventTarget = event.target;
+
+    this.events[type].forEach(ev => {
+      // A bit hacky but this allows using the same api to bind to window for key, resize and scroll events
+      const target = ev.selector === 'window' ? window : eventTarget.closest(ev.selector);
+
+      if (target) {
+        ev.func(target, event);
```

Comment:
It is a little bit confusing that event handler uses non standard parameters, so instead of just having event (or at least having event as first param) it has target first and then event.

Reply:
I've switched these around, I originally put target first, as more of the code seemed to use target (`event` was optional) but it makes more sense to follow the standard of `event` first, as it's a more complete resource.

We could technically get rid of the `target`, but as we do the lookup to closest on `trigger`, we may as well pass it through to save doing it again in most cases.